nexus-sdk/src/main/java/io/nexusrpc/Operation.java

```diff
@@ -6,16 +6,16 @@
 import java.lang.annotation.Target;
 
 /**
- * Operations are interface methods defined on {@link Service} interfaces. This annotation is not
- * inherited, so it must be on any overridden signature in any sub-interface with the same values or
- * an error will occur. Similarly, an error will occur if an implementation has multiple operation
- * definitions of the same name with different signatures.
+ * Operations are interface methods defined on {@link Service} interfaces.
  *
- * <p>Return type is {@link java.util.concurrent.Future} for async operations, or any other value
- * for synchronous operations. The operation may accept no more than one parameter. All types must
- * be serializable.
+ * <p>This annotation is not inherited, so it must be on any overridden signature in any
```

Comment:
Is there a reason this is not inherited like other workflow annotations?

Reply:
Class inheritance works, but I was not aware we inherited other workflow annotations (IMO if that's intentional we should consider adding `@Inherited` on them even if in reflection we do reference the explicitly declared ones)

So to confirm, in Java SDK, what happens here:

```java
interface Foo {
  @SignalMethod
  Object doSomething(String bar);
}

@Workflow
interface Bar extends Foo {
  @Override
  String doSomething(String bar);
}
```

Does that get the signal method? In .NET and Python at least we did not do that to avoid a diamond problem and a clarity problem. Also unlike Java SDK (I think?), we disallow _any_ method declaration on a service interface that is not marked as an operation, again for clarity/consistency reasons. This is a contract, not to be implemented (again unlike Java SDK).

Reply:
Overloading is not allowed I believe, but you can create a workflow interface by inheriting from other interfaces like what is described https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/workflow/WorkflowInterface.html

Reply:
Method/class inheritance does work like that with Nexus here. What "This annotation is not inherited" means is not that class/method inheritance does not work, it means if you redefine it in a subclass, you must also provide the annotation. It is specifically discussing the Java concept of _annotation_ inheritance, not method/class inheritance.